### DISCUSSION TITLE
Equivalent of CPack for cross-platform packaging?

### DISCUSSION THREAD

**Discussion by matty0ung** [mesonbuild/meson#11090] [Q&A]
I help maintain an open-source cross-platform (using Qt) desktop application.  Currently we build it with CMake and CPack.  I have been looking at using Meson instead.  Compilation etc seems to work fine but I can't see much advice about packaging.   I found advice for [Mac](https://mesonbuild.com/Creating-OSX-packages.html) and [Linux Binaries](https://mesonbuild.com/Creating-Linux-binaries.html).  But I don't see any how-to for creating a Windows installer (or for creating RPM or deb packages for Linux).

For the avoidance of doubt, I'm looking to create on each platform (Windows / Mac / Linux) a free-standing download that a non-technical user can install.  In other words, on Windows and Mac, I can't assume that the user will already have a particular package management system installed.

What's the best approach?

**Comment by jpakkane** [comment]
One simple way is to do the same "install into a prefix dir and package that" thing with [Msicreator](https://github.com/jpakkane/msicreator/), which uses WiX behind the scenes.
